src/zenml/integrations/kubeflow/metadata_stores/kubeflow_metadata_store.py

```diff
@@ -60,12 +88,195 @@ def get_tfx_metadata_config(
         metadata_store_pb2.MetadataStoreClientConfig,
     ]:
         """Return tfx metadata config for the kubeflow metadata store."""
+        connection_config = metadata_store_pb2.MetadataStoreClientConfig()
         if inside_kfp_pod():
-            connection_config = metadata_store_pb2.MetadataStoreClientConfig()
             connection_config.host = os.environ["METADATA_GRPC_SERVICE_HOST"]
             connection_config.port = int(
                 os.environ["METADATA_GRPC_SERVICE_PORT"]
             )
-            return connection_config
         else:
-            return super().get_tfx_metadata_config()
+            if not self.is_running:
+                raise RuntimeError(
+                    "The KFP metadata daemon is not running. Please run the "
+                    "following command to start it first:\n\n"
+                    "    'zenml metadata-store up'\n"
+                )
+            connection_config.host = self.host
+            connection_config.port = self.port
+        return connection_config
+
+    @property
+    def kfp_orchestrator(self) -> KubeflowOrchestrator:
+        """Returns the Kubeflow orchestrator in the active stack."""
+        repo = Repository(skip_repository_check=True)  # type: ignore[call-arg]
+        return cast(KubeflowOrchestrator, repo.active_stack.orchestrator)
+
+    @property
+    def kubernetes_context(self) -> str:
+        """Returns the kubernetes context to the cluster where the Kubeflow
+        Pipelines services are running."""
+        return self.kfp_orchestrator.kubernetes_context
+
+    @property
+    def root_directory(self) -> str:
+        """Returns path to the root directory for all files concerning
+        this KFP metadata store.
+
+        Note: the root directory for the KFP metadata store is relative to the
+        root directory of the KFP orchestrator, because it is a sub-component
+        of it.
+        """
+        return os.path.join(
+            self.kfp_orchestrator.root_directory,
+            "metadata-store",
+            str(self.uuid),
+        )
+
+    @property
+    def _pid_file_path(self) -> str:
+        """Returns path to the daemon PID file."""
+        return os.path.join(self.root_directory, "kubeflow_daemon.pid")
+
+    @property
+    def _log_file(self) -> str:
+        """Path of the daemon log file."""
+        return os.path.join(self.root_directory, "kubeflow_daemon.log")
+
+    @property
+    def is_provisioned(self) -> bool:
+        """If the component provisioned resources to run locally."""
+        return fileio.exists(self.root_directory)
+
+    @property
+    def is_running(self) -> bool:
+        """If the component is running locally."""
+        if sys.platform != "win32":
+            if not check_if_daemon_is_running(self._pid_file_path):
+                return False
+        else:
+            pass
+        return True
```

Comment:
Just a comment here: what are the advantages of writing it in this way, vs just returning as part of the `else:` condition?

Reply:
I'm gonna take a stab at replying with what I guess is the reason: This `return True` at the end essentially catches the `else` conditions of both if conditions in this function which is why it's at the end I think.

Reply:
No particular advantage, I just have to do something in the "else" clause to keep mypy from complaining on Windows.